usecases/replica/transport.go

```diff
@@ -157,22 +163,43 @@ type Client interface {
 
 // RClient is the client used to read from remote replicas
 type RClient interface {
-	FindObject(_ context.Context, host, index, shard string,
+	// FetchObject fetches one object
+	FetchObject(_ context.Context, host, index, shard string,
 		id strfmt.UUID, props search.SelectProperties,
-		additional additional.Properties) (*storobj.Object, error)
+		additional additional.Properties) (objects.Replica, error)
 
 	Exists(_ context.Context, host, index, shard string, id strfmt.UUID) (bool, error)
 
-	MultiGetObjects(_ context.Context, host, index, shard string,
-		ids []strfmt.UUID) ([]*storobj.Object, error)
+	// FetchObjects fetches objects specified in ids list.
+	FetchObjects(_ context.Context, host, index, shard string,
+		ids []strfmt.UUID) ([]objects.Replica, error)
+
+	// OverwriteObjects conditionally updates existing objects.
+	OverwriteObjects(_ context.Context, host, index, shard string,
+		_ []*objects.VObject) ([]RepairResponse, error)
+
+	// DigestObjects finds a list of objects and returns a compact representation
+	// of a list of the objects. This is used by the replicator to optimize the
+	// number of bytes transferred over the network when fetching a replicated
+	// object
+	DigestObjects(ctx context.Context, host, index, shard string,
+		ids []strfmt.UUID) ([]RepairResponse, error)
 }
 
 type RepairResponse struct {
 	ID         string // object id
-	Version    uint64 // sender's current version of the object
+	Version    int64  // sender's current version of the object
```

Comment:
Does it make sense to have a negative version?

Reply:
Logically yes
But practically int64 is better because it can be natively serialised into json we negative values might be also useful (e.g. deletion)

Reply:
take for example function slice  `len()` it returns int not uint even though the length is always >= 0

Reply:
Ok, understood.